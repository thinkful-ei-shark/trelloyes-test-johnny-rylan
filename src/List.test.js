import React from 'react';
import ReactDOM, { unmountComponentAtNode } from 'react-dom';
import renderer from 'react-test-renderer';
import STORE from './STORE';

import List from './Card';

describe('List component', () => {
  // smoke test
  it('Should render Section without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<List />, div);
    unmountComponentAtNode(div);
  });
  // snapshot test
  it('renders List UI as expected', () => {
    const lists = STORE.lists.map(list => {
      return <List
        key={list.id}
        header={list.header}
        cards={list.cardIds.map(id => STORE.allCards[id])}
      />
    })
    const tree = renderer
      .create(lists)
      .toJSON()
    expect(tree).toMatchSnapshot();

  })
});