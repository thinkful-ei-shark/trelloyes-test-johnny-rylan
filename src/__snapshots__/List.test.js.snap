// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders List UI as expected 1`] = `
Array [
  <div
    className="Card"
  >
    <button
      type="button"
    >
      delete
    </button>
    <h3 />
    <p />
  </div>,
  <div
    className="Card"
  >
    <button
      type="button"
    >
      delete
    </button>
    <h3 />
    <p />
  </div>,
  <div
    className="Card"
  >
    <button
      type="button"
    >
      delete
    </button>
    <h3 />
    <p />
  </div>,
  <div
    className="Card"
  >
    <button
      type="button"
    >
      delete
    </button>
    <h3 />
    <p />
  </div>,
]
`;
